src/main/kotlin/pl/edu/uj/ii/ksi/mordor/persistence/managers/MetadataManager.kt

```diff
@@ -0,0 +1,21 @@
+package pl.edu.uj.ii.ksi.mordor.persistence.managers
+
+import java.io.File
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileContent
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileEntry
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileMetadata
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileThumbnail
+
+interface MetadataManager {
+    fun extractAndSaveMetadata(file: File): FileMetadata
```

Comment:
SRP
Client doesn't need to know what happens inside. I suggest renaming it to `saveFileMetadata` and do not return it.

Reply:
How will the client find this specific metadata later, i.e. to create new FileEntry?